resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java

```diff
@@ -36,15 +37,16 @@ public String serialize() {
   }
 
   // Print the acknowledgement received from IoT Hub for the telemetry message sent.
-  private static class EventCallback implements IotHubEventCallback {
-    public void execute(IotHubStatusCode status, Object context) {
-      System.out.println("IoT Hub responded to message with status: " + status.name());
-
-      if (context != null) {
-        synchronized (context) {
-          context.notify();
+  private static class EventCallback implements MessageSentCallback {
+    @Override
+    public void onMessageSent(Message sentMessage, IotHubClientException exception, Object callbackContext) {
```

Comment:
is there any need for sentMessage?

Reply:
In our sample, no - but this is an implementation from `MessageSentCallback` - so the data will likely come from the SDK regardless - the user can check additional message properties or data in this callback.